lib/src/working_copy.rs

```diff
@@ -293,6 +293,19 @@ impl TreeState {
         let git_repo_dir = tempfile::tempdir().unwrap();
         let mut git_repo_options = RepositoryInitOptions::new();
         git_repo_options.workdir_path(&self.working_copy_path);
+        // On Windows, Repository::init_opts creates a ".git" file in the working copy,
+        // and that ".git" file will make the next Repository::init_opts fail with
+        // "Permission Denied". So let's remove the ".git" file automatically.
+        #[cfg(windows)]
```

Comment:
The `.git` file is actually created and left there on all platforms. It's https://github.com/libgit2/libgit2sharp/issues/1716 (which seems to be in libgit2 itself, not just in the .NET wrapper). Could you drop the `#[cfg(windows)]` and update the comment (but still mention that it's only a real problem on Windows)?

Thanks for fixing the bug, btw :) I added .git to my .gitignore a long time ago and haven't thought much about it, but it's definitely a rough edge that's nice to fix.

Reply:
Makes sense. I didn't run it on Linux to check, so I thought it might be a Windows only behavior.